src/x86/intel_edison_fab_c.c

```diff
@@ -373,8 +374,17 @@ mraa_intel_edison_aio_init_post(mraa_aio_context dev)
 mraa_result_t
 mraa_intel_edison_pwm_enable_pre(mraa_pwm_context dev, int enable) {
     // PWM 0% duty workaround: update state array
-    // if someone first ran write(0) and then enable(1).
-    if ((pwm_disabled[dev->pin] == 1) && (enable == 1)) { pwm_disabled[dev->pin] = 0; }
+    if (enable == 1) {
+        pwm_disabled[dev->pin] = 0;
+    } else {
+        pwm_disabled[dev->pin] = 1;
+        // Set duty cycle to 0 before disable pwm.
+        // Edison PWM output stuck at high if disabled during ON period
+        mraa_pwm_pulsewidth_us(dev, 0);
```

Comment:
This means the user's duty setting will be lost after they run `enable(0)` or `write(0)`, which will break existing code relying on the setting preservation. I'd say we should avoid that at all costs.